armotypes/vulnerabilitytypes.go

```diff
@@ -9,11 +21,175 @@ type VulnerabilityJobParams struct {
+	ClusterName      string                     `json:"clusterName"`
+	ClusterShortName string                     `json:"clusterShortName"`
+	CustomerGUID     string                     `json:"customerGUID"`
+	SeverityStats    map[string]SeverityDetails `json:"severityStats"`
+	Paths            []string                   `json:"paths"`
+}
+
 type Vulnerability struct {
-	Name          string   `json:"name"`
-	Severity      string   `json:"severity"`
-	SeverityScore int      `json:"severityScore"`
-	IsRCE         bool     `json:"isRCE"`
-	Links         []string `json:"links"`
-	Description   string   `json:"description"`
+	Id             string              `json:"id"`
+	Name           string              `json:"name"`
+	SeverityScore  int                 `json:"severityScore"`
+	IsRCE          bool                `json:"isRCE"`
+	Links          []string            `json:"links"`
+	Description    string              `json:"description"`
+	Exploitable    string              `json:"exploitable"`
+	ComponentsInfo []CveComponentsInfo `json:"componentsInfo"`
+	IsFixable      bool                `json:"isFixable"`
+	CvssInfo       []CvssInfo          `json:"cvssInfo"`
+	EpssInfo       EpssInfo            `json:"epssInfo"`
+	CisaKevInfo    CisaKevInfo         `json:"cisaKevInfo"`
+}
+
+type CveComponentsInfo struct {
+	Name           string   `json:"name"`
+	Version        string   `json:"version"`
+	PackageType    string   `json:"packageType"`
+	Paths          []string `json:"paths"`
+	AffectedImages []string `json:"affected_images"`
+}
+type CvssInfo struct {
+	Vector              string                 `json:"vector"`
+	Version             string                 `json:"version"`
+	Source              string                 `json:"source"`
+	BaseScore           int                    `json:"baseScore"`
+	ExploitabiltiyScore int                    `json:"exploitabiltiyScore"`
+	ImpactScore         int                    `json:"ImpactScore"`
+	ExploitabilityInfo  CvssExploitabilityInfo `json:"exploitabilityInfo"`
+	ImpactInfo          CvssImpactInfo         `json:"impactInfo"`
+}
+
+type CvssExploitabilityInfo struct {
+	AttackVector       string `json:"attackVector"`
+	AttackComplexity   string `json:"attackComplexity"`
+	PrivilegesRequired string `json:"privilegesRequired"`
+	UserInteraction    string `json:"userInteraction"`
+	Scope              string `json:"scope"`
+}
+
+type CvssImpactInfo struct {
+	Confidentiality string `json:"confidentiality"`
+	Integrity       string `json:"integrity"`
+	Availability    string `json:"availability"`
+}
+
+type EpssInfo struct {
+	Epss       float64   `json:"epss"`
+	Percentile float64   `json:"percentile"`
+	Date       time.Time `json:"date"`
+}
+
+type CisaKevInfo struct {
+	// todo : verify tooltip and additional feilds
+	Date time.Time `json:"date"`
+}
+
+// EnrichCVSSData enriches the CVSS data by parsing the CVSS vector and updating the exploitability field
+func EnrichCVSSData(vuln *Vulnerability) {
```

Comment:
I am ot sure  armo types is a good place for these functions ?

Reply:
are we sure that enrichment will happen only in BL? I thought about having it in common place